### ISSUE TITLE
Oscilloscope crashes when attempting to turn ON

### ISSUE THREAD

**Issue by dondonddon** [rogeriobego/oscilloscope-arduino-processing#9]
Uploading the Arudino code and also downloading the Java app for the Oscilloscope was straightforward and this project seems very exciting.  I really hope to get it to work.   Quite impressive.  

An online video shows that the proper process to run the oscilloscope is to perform the following steps;
1. Upload the Arduino code to the Arduino.  Note the serial speed and COM port.  
2. Without the Arduino IDE running, plug in the configured Arduino, and note the COM port in Device Manager (confirm).
3. Open the Oscilloscope app.  At the top, select the proper COM port and serial speed.

However, when I then attempt to turn the "OFF" to "ON", the java app crashes.   Note that prior to the crash, the DEMOs all work fine.  The crash also results in an error file being produced.  See attached file. 

As noted in the attached file, I am using a 64 bit Windows 11 laptop and the latest Java version (JRE 1.8.0).    

[hs_err_pid7260.log](https://github.com/rogeriobego/oscilloscope-arduino-processing/files/9862956/hs_err_pid7260.log)

**Comment by Muhammadyusuf96** [comment]
same

**Comment by Muhammadyusuf96** [comment]
It worked after I rebuild it

**Comment by dondonddon** [comment]
Yes!  Thank you.   I went ahead and downloaded the Processing application so that I could run the oscilloscope_4ch.pde file.  That made the Windows application run and I could now correctly configure it for the serial rate and COM port.  So it is now working!  

But it does require me each time I use the application to bring up the Processing app and execute the oscilloscope_4ch.pde file (since I can't save it as an .exe file), and then run the Arduino sketch on the nano, and finally to synch them for the proper serial rate and COM port.   

Thanks again.  It is good to know that it works!

**Comment by Muhammadyusuf96** [comment]
Why you can't save as exe?

**Comment by dondonddon** [comment]
Actually, I found out yesterday that you CAN save the .exe file.   After opening the Processing app and running the sketch, one can File..Export Application and the resulting executable will be saved:   source.exe    One can then use that executable without needing to open the Processing app again.  
I put together a simple 2-transistor audio oscillator and experimented briefly with trying to feed the small audio signal into the A0 (channel zero). From the attached speaker, it was obvious that I should be seeing something like a periodic signal in the 500 - 1000 hertz range, but the oscillosocpe  didn't appear to be able to keep up with that signal.   I wonder if others have had similar results? I expect that this system won't support high frequencies.  We know it works for 50 hz from the demo signal that A9 and A10 provide so I expected it would also work up to at least 1000 hz.

**Comment by Muhammadyusuf96** [comment]
I think it's because Arduino is so noisy and slow

**Comment by dondonddon** [comment]
Bottom line for this issue is that it is resolved.     
Regarding the performance,  I decided to try  a different approach today.  I used a TONE app on my iPhone to generate a pure tone at various frequencies (200 through 2500 hz) and then used a MAX4466 mic/amp to listen to that signal and feed it through to the A0 ADC.   Attached are some of the test results showing that the oscilloscope is generally able to detect the main tone.  This seemed to work all the way up to 2500 hz.  Not pretty, but reasonable.  The MAX4466 is known to have lots of noise.  
The photos below are for 2500 hz, 400 hz, and 1000 hz.   
![2500_hz](https://github.com/rogeriobego/oscilloscope-arduino-processing/assets/72356730/596d0a24-3b10-47e4-bec2-192ef33f03e2)
![400_hz_tone](https://github.com/rogeriobego/oscilloscope-arduino-processing/assets/72356730/292522b6-caeb-4307-8909-30a0093ea16c)
![1000_hz_tone](https://github.com/rogeriobego/oscilloscope-arduino-processing/assets/72356730/e7086138-3a49-430d-9b4a-07922c5f07aa)
